test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/integration/ProcessInstanceMigrationExternalDatabaseIntegrationTest.java

```diff
@@ -0,0 +1,210 @@
+        Assume.assumeTrue(isExternalDatabaseAllocated());
+        try {
+            deploymentScenario = deploymentScenarioFactory.getKieServerWithExternalDatabaseScenarioBuilder()
+                                                          .withInternalMavenRepo(true)
+                                                          .withContainerDeployment(KIE_CONTAINER_DEPLOYMENTS)
+                                                          .withProcessMigrationDeployment()
+                                                          .build();
+
+
+            KjarDeploymentScenarioListener.addKjarDeployment(deploymentScenario, OLD_KJAR, NEW_KJAR);
+        } catch (UnsupportedOperationException ex) {
+            Assume.assumeFalse(ex.getMessage().startsWith("Not supported"));
+        }
+        deploymentScenario.setLogFolderName(ProcessInstanceMigrationExternalDatabaseIntegrationTest.class.getSimpleName());
+        ScenarioDeployer.deployScenario(deploymentScenario);
+
+    }
+
+    @Before
+    public void setUp() throws KeyManagementException, NoSuchAlgorithmException, KeyStoreException {
+        logger.info("URL " + deploymentScenario.getKieServerDeployment().getUrl());
+        kieServicesClient = KieServerClientProvider.getKieServerClient(deploymentScenario.getKieServerDeployment());
+        processServicesClient = KieServerClientProvider.getProcessClient(deploymentScenario.getKieServerDeployment());
+
+        kieServerId = kieServicesClient.getServerInfo().getResult().getServerId();
+        client = HttpClientBuilder.create().setDefaultCredentialsProvider(getBasicAuth()).build();
+    }
+
+    @Test
+    public void testProcessInstanceMigrationIntegration() {
+        // Given
+        startProcesses();
+
+        // When
+        try {
+            createMigration();
+        } catch (IOException | JAXBException e) {
+            throw new RuntimeException("Process instance migration failed", e);
+        }
+
+        // Then
+        List<ProcessInstance> instances = processServicesClient.findProcessInstances(OLD_CONTAINER_ID, 0, 10);
+        Assertions.assertThat(instances).hasSize(1);
+        Assertions.assertThat(instances.get(0).getId()).isIn(instancesIds);
+
+        instances = processServicesClient.findProcessInstances(NEW_CONTAINER_ID, 0, 10);
+        Assertions.assertThat(instances).hasSize(1);
+        Assertions.assertThat(instances.get(0).getId()).isIn(instancesIds);
+    }
+
+    private Migration createMigration() throws IOException, JAXBException {
+        Plan plan = createPlan();
+        MigrationDefinition migrationDefinition = new MigrationDefinition();
+        migrationDefinition.setPlanId(plan.getId());
+        migrationDefinition.setKieServerId(kieServerId);
+        migrationDefinition.setProcessInstanceIds(Arrays.asList(instancesIds.get(0)));
+        migrationDefinition.setExecution(new Execution().setType(Execution.ExecutionType.SYNC));
+
+        HttpPost post = preparePost("/migrations");
```

Comment:
Same comment as below: this is too low level for me. Maybe we could follow the same we have for kieserver or workbench: adding a ProcessInstanceClient with a method startMigration.
What do you think?